src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -184,6 +188,8 @@ public KeyRangeIterator.Builder getIndexQueryResults(Collection<Expression> expr
 
         QueryViewBuilder.QueryView queryView = queryViewBuilder.build();
 
+        maybeTriggerGuardrails(queryView);
```

Comment:
So I think the reason you're seeing LEAKs in the new tests is that this needs to happen inside the `try` below. If we don't do that, we won't release the `SSTableIndex`es when we hit the failure threshold.

(Just verified locally on my machine that your tests are clean w/ this change.)

Reply:
hah great catch!